spec/unit/resource_spec.rb

```diff
@@ -189,12 +188,16 @@
     let(:prior_resource) do
       prior_resource = Chef::Resource.new("funk")
       prior_resource.source_line
-      prior_resource.allowed_actions << :funkytown
```

Comment:
But this original behavior will continue to not work right, even after changing allowed_actions. My first thought is that no subclass  should be directly modifying that list anyway.